File: java_solutions/1071.greatest-common-divisor-of-strings.java
/*
 * @lc app=leetcode id=1071 lang=java
 *
 * [1071] Greatest Common Divisor of Strings
 */

import java.math.*;
class Solution {
    public String gcdOfStrings(String s1, String s2) {
        return (s1 + s2).equals(s2 + s1) ? s1.substring(0, BigInteger.valueOf(s1.length()).gcd(BigInteger.valueOf(s2.length())).intValue()) : "";
    }
}

File: kotlin_solutions_1/1071.greatest-common-divisor-of-strings.kt
/*
 * @lc app=leetcode id=1071 lang=kotlin
 *
 * [1071] Greatest Common Divisor of Strings
 */

class Solution {
    fun gcdOfStrings(s1: String, s2: String): String = if (s1 + s2 == s2 + s1) s1.substring(0, java.math.BigInteger.valueOf(s1.length.toLong()).gcd(java.math.BigInteger.valueOf(s2.length.toLong())).toInt()) else ""
}